src/components/Board/index.spec.js

```diff
@@ -354,6 +354,26 @@ describe('<Board />', () => {
         })
       })
     })
+
+    describe('when the component receives a "renderLaneHeader" prop', () => {
```

Comment:
I think this context should be inside of the `when it receives the "allowAddLane" and "onNewLane" prop` context. 

`allowAddLane` should be considered even when the user pass a lane adder template.

Please consider that and adjust your specs. You might need to cover some other contexts too.

```suggestion
    describe('when the component receives a "renderLaneAdder" prop', () => {
```